upass/__main__.py

```diff
@@ -274,6 +276,66 @@ def copy_selected(self, originator):
         if pathg in self.passwords:
             self.call_pass(originator, (path, True, False))
 
+    def generate_password(self, originator, args=None):
+        if args:
+            path, length, symbols, force = args
+        else:
+            path, length, symbols, force = '', 16, True, False
+        if not path and self.current != '.':
+            path = self.current + '/'
+        self.set_header('GENERATE PASSWORD')
+        self._clear_box()
+        self.generate_password_path = urwid.Edit(("highlight", "Passowrd Name: "), path)
+        self.generate_password_length = urwid.IntEdit("Length: ", default=length)
+        self.generate_password_symbols = urwid.CheckBox("Symbols", state=symbols)
+        self.generate_password_force = urwid.CheckBox("Overwrite existing passwords with that name", state=force)
+        self.box.body.append(self.generate_password_path)
+        self.box.body.append(self.generate_password_length)
+        self.box.body.append(self.generate_password_symbols)
+        self.box.body.append(self.generate_password_force)
+        self.box.body.append(ActionButton('GENERATE', self.call_generate))
+        self.box.body.append(BackButton('BACK', self.load_dispatch, self.current, self))
+
+    def call_generate(self, originator):
+        gargs = ['pass', 'generate']
+        symbols = self.generate_password_symbols.state
+        force = self.generate_password_force.state
+        if not symbols:
+            gargs.append('-n')
+        if force:
+            gargs.append('-f')
+        path = self.generate_password_path.text[len("Password Name: "):]
+        length = str(self.generate_password_length.value())
+        gargs += [path, length]
+        self._clear_box()
+        self.box.body.append(urwid.AttrMap(
+            urwid.Text(
+                'Generating "{0}": (length: {1}, symbols: {2}, overwrite: {3})'.format(path, length, symbols, force)
+            ), 'highlight')
+        )
+        self.loop.draw_screen()
+        if path.endswith('/'):
+            errormsg = 'Invalid key path "{0}". Please enter a valid password name.'.format(path)
+        elif os.path.exists(os.path.join(self.home, path + '.gpg')) and not force:
+            errormsg = ('Password with name "{0}" already exists. '
+                        'Please select the option to overwrite the existing password to generate a new password '
+                        'for this entry.'.format(path))
+        else:
+            p = subprocess.Popen(gargs, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            stdout, errormsg = p.communicate()
```

Comment:
a 5-second timeout would be a good idea. also, perhaps we could display the password if it’s successfully generated?

Reply:
Thanks, I will put the longer timeout back in. For the second point, currently the password is loaded after being created, so the user can choose to display it from there (or copy it, if they prefer). Do you think it's better to go straight to showing the new password?

Reply:
I think yes, so we work the same way `pass generate` does (without requiring a password).

Reply:
Makes sense, will submit a new commit soon.